drivers/vhdutil.py

```diff
@@ -115,7 +120,7 @@ def getVHDInfo(path, extractUuidFunction, includeParent=True):
             vhdInfo.parentUuid = extractUuidFunction(fields[nextIndex])
         nextIndex += 1
     vhdInfo.hidden = int(fields[nextIndex].replace("hidden: ", ""))
-    vhdInfo.sizeAllocated = int(fields[nextIndex+1])
+    vhdInfo.sizeAllocated = convertAllocatedSizeToBytes(int(fields[nextIndex+1]))
```

Comment:
is the block size a fixed value here?

Reply:
it technically is not but in practice all VHDs created in XenServer are using 2MB allocation blocks. The allocation block size is a property in the VHD header.

Reply:
> it technically is not but in practice all VHDs created in XenServer are using 2MB allocation blocks.
Thanks for point this out @MarkSymsCtx , then fixed value work.
from a compatibility perspective, I think calculating it with size in VHD metadata would be better. 
Just a thought for future improvements.